lib/practice_exercises.rb

```diff
@@ -1,13 +1,63 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n), where n is the length of the string
+# The loop in this method get ran 1/2n
+# dropping the coefficient will be O(n)
+# Space Complexity: 0(1)
+# uses constant amount of memory
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  # replace all non letters with empty string
+  new_string = string.upcase.gsub(/[^A-Za-z]/, '')
+
+  # a starts at index 0
+  # b is the last index of the string
+  a = 0
+  b = new_string.length - 1
+
+  # compares each letter starting at the ends of the string
+  # for every loop, a increases by one and b decreases by one
+  while a <= b
+    if new_string[a] != new_string[b]
+      return false
+    end
+
+    a += 1
+    b -= 1
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n*m)
+# n is the length of the shortest string
+# m is the length of the strings array
+# Space Complexity: O(1)
+# uses constant amount of memory
 def longest_prefix(strings)
```

Comment:
👍 You are building a prefix so your space complexity is O(m) where m is the shortest string.